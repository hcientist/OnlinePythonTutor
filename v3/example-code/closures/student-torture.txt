# Example to torture students
# from UC Berkeley CS61a
# (only works in Python 3)

def f(t):
    def g(t):
        def h():
            nonlocal t
            t += 1
        return h, lambda: t
    h, gt = g(0)
    return h, gt, lambda: t

h, gt, ft = f(0)
ft(), gt()
h()
ft(), gt()
