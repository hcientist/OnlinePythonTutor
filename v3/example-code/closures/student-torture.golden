{
  "code": "# Example to torture students\n# from UC Berkeley CS61a\n# (only works in Python 3)\n\ndef f(t):\n    def g(t):\n        def h():\n            nonlocal t\n            t += 1\n        return h, lambda: t\n    h, gt = g(0)\n    return h, gt, lambda: t\n\nh, gt, ft = f(0)\nft(), gt()\nh()\nft(), gt()\n", 
  "trace": [
    {
      "exception_msg": "SyntaxError: invalid syntax (<string>, line 8)", 
      "line": 8, 
      "event": "uncaught_exception", 
      "offset": 22
    }
  ]
}
